### PR TITLE
Add `firstInterimResponseStart` and `deliveryType` to PerformanceResourceTiming

### PR DESCRIPTION
**Pull request by tunetheweb**
### Description

Add `deliveryType` and `firstInterimResponseStart` to PerformanceResourceTiming

I also added experimental status to `renderBlockingStatus` as currently Chromium only.

### Motivation

New additions landing in Chrome

### Additional details

- firstInterimResponseStart in Chrome 115 - https://chromestatus.com/feature/5086730938482688
- deliveryType in Chrome 117 - https://chromestatus.com/feature/6347141115543552

### Related issues and pull requests

These have already been added to browser-compat-data by @Elchi3 in https://github.com/mdn/browser-compat-data/pull/20518 and https://github.com/mdn/browser-compat-data/pull/20519

### CODE REVIEW

File: files/en-us/web/api/performanceresourcetiming/deliverytype/index.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+title: "PerformanceResourceTiming: deliveryType property"
+short-title: deliveryType
+slug: Web/API/PerformanceResourceTiming/deliveryType
+page-type: web-api-instance-property
+browser-compat: api.PerformanceResourceTiming.deliveryType
+---
+
+{{APIRef("Performance API")}}
+
+The **`deliveryType`** read-only property is a string representing web platform feature that exposes information about how the resource was delivered. For example, resources which were delivered from the cache.
```
Comment (sideshowbarker) [inline files/en-us/web/api/performanceresourcetiming/deliverytype/index.md:11]:
```suggestion
The **`deliveryType`** read-only property is a string indicating how the resource was delivered — for example, indicating whether the resource was delivered from the cache.
```

File: files/en-us/web/api/performanceresourcetiming/deliverytype/index.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+## Value
+
+The `deliveryType` property can have the following values:.
+
+- `cache`
+  - : If the request was retrieved from the cache.
+- `""` (empty string)
+  - : The empty string is returned otherwise.
+
+This is expected to be expanded by future updates to this specification, e.g. to describe consuming preloaded resources and prefetched navigation requests.
```
Comment (sideshowbarker) [inline files/en-us/web/api/performanceresourcetiming/deliverytype/index.md:22]:
```suggestion
The `deliveryType` property can have the following values:.

- `cache`
  - : If the resource was retrieved from the cache
- `""` (empty string)
  - : If none of the other defined delivery types

This set of delivery types is expected to be expanded in the future — for example, to indicate preloaded resources and to indicate prefetched navigation requests.
```

File: files/en-us/web/api/performanceresourcetiming/deliverytype/index.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+## Examples
+
+### Filtering resources
+
+The `deliveryType` property can be used to get specific resource timing entries only. For example, only those that were cached.
```
Comment (sideshowbarker) [inline files/en-us/web/api/performanceresourcetiming/deliverytype/index.md:28]:
```suggestion
The `deliveryType` property can be used to get specific resource timing entries only; for example, only those that were cached.
```

File: files/en-us/web/api/performanceresourcetiming/deliverytype/index.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+Example using a {{domxref("PerformanceObserver")}}, which notifies of new `resource` performance entries as they are recorded in the browser's performance timeline. Use the `buffered` option to access entries from before the observer creation.
```
Comment (sideshowbarker) [inline files/en-us/web/api/performanceresourcetiming/deliverytype/index.md:30]:
```suggestion
The following example uses a {{domxref("PerformanceObserver")}} which notifies of new `resource` performance entries as they are recorded in the browser's performance timeline. The `buffered` option is used for accessing entries from before the observer creation.
```

File: files/en-us/web/api/performanceresourcetiming/deliverytype/index.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  const cachedResources = list.getEntries().filter((entry) => {
+    return entry.deliveryType === "cache";
+  });
+  console.log(cachedResources);
+});
+
+observer.observe({ type: "resource", buffered: true });
+```
+
+Example using {{domxref("Performance.getEntriesByType()")}}, which only shows `resource` performance entries present in the browser's performance timeline at the time you call this method:
```
Comment (sideshowbarker) [inline files/en-us/web/api/performanceresourcetiming/deliverytype/index.md:43]:
```suggestion
The following example uses {{domxref("Performance.getEntriesByType()")}}, which only shows `resource` performance entries present in the browser's performance timeline at the time you call the method.
```

File: files/en-us/web/api/performanceresourcetiming/firstinterimresponsestart/index.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+title: "PerformanceResourceTiming: firstInterimResponseStart property"
+short-title: firstInterimResponseStart
+slug: Web/API/PerformanceResourceTiming/firstInterimResponseStart
+page-type: web-api-instance-property
+browser-compat: api.PerformanceResourceTiming.firstInterimResponseStart
+---
+
+{{APIRef("Performance API")}}
+
+The **`firstInterimResponseStart`** read-only property returns a {{domxref("DOMHighResTimeStamp","timestamp")}} immediately after the browser receives the first byte of the interim 1xx response (e.g. 100 continue or 103 early hints) from the server.
```
Comment (sideshowbarker) [inline files/en-us/web/api/performanceresourcetiming/firstinterimresponsestart/index.md:11]:
```suggestion
The **`firstInterimResponseStart`** read-only property returns a {{domxref("DOMHighResTimeStamp","timestamp")}} immediately after the browser receives the first byte of the interim 1xx response (e.g. 100 Continue or 103 Early Hints) from the server.
```

File: files/en-us/web/api/performanceresourcetiming/firstinterimresponsestart/index.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+- A {{domxref("DOMHighResTimeStamp")}} immediately after the browser receives the first interim bytes of the response from the server.
+- `0` if the resource was instantaneously retrieved from a cache.
+- `0` if the resource is a cross-origin request and no {{HTTPHeader("Timing-Allow-Origin")}} HTTP response header is used.
+
+## Examples
+
+### Measuring request time
+
+The `firstInterimResponseStart` and {{domxref("PerformanceResourceTiming.requestStart", "requestStart")}} properties can be used to measure how long the request takes to send an interim response.
```
Comment (sideshowbarker) [inline files/en-us/web/api/performanceresourcetiming/firstinterimresponsestart/index.md:27]:
```suggestion
The `firstInterimResponseStart` and {{domxref("PerformanceResourceTiming.requestStart", "requestStart")}} properties can be used to measure how long it takes to the browser to receive an interim response after the sending the request.
```

File: files/en-us/web/api/performanceresourcetiming/firstinterimresponsestart/index.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+```js
+const request = entry.firstInterimResponseStart - entry.requestStart;
+```
+
+Example using a {{domxref("PerformanceObserver")}}, which notifies of new `resource` performance entries as they are recorded in the browser's performance timeline. Use the `buffered` option to access entries from before the observer creation.
```
Comment (sideshowbarker) [inline files/en-us/web/api/performanceresourcetiming/firstinterimresponsestart/index.md:33]:
```suggestion
The following example uses a {{domxref("PerformanceObserver")}} to notify of new `resource` performance entries as they are recorded in the browser's performance timeline. The `buffered` option is used for accessing entries from before the observer creation.
```

**Review by sideshowbarker — APPROVED** [review]

**Comment by tunetheweb** [comment]
Thanks @sideshowbarker for the thorough review! Promise to do better next time.

**Pull request merged**